### ISSUE TITLE
Non EntryPoint Startup throws generic NullReferenceException for WebApplicationFactory

### ISSUE THREAD

**Issue by caseylobrien** [aspnet/Mvc#8250]
#### Is this a Feature request:
Throw a more useful error message for using a non `System.Reflection.Assembly.EntryPoint` as the generic passed into the WebApplicationFactory<TEntryPoint> Ideally the Exception would be thrown in a better place (and sooner) then `WebApplicationFactory.CreateClient`

#### Steps to reproduce:
https://github.com/caseylobrien/ClassLibraryStartup

#### Description of the problem:
[Original Issue](https://github.com/aspnet/Hosting/issues/1517#issuecomment-411917607)

TLDR - Spin up an in memory server following this [guide](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.1) for integration tests. The structure of the solution is not typical, the program.cs references a Startup.cs that is in a class library

#### `Microsoft.AspNetCore.Mvc.Testing` Version="2.1.1"

**Comment by caseylobrien** [comment]
@javiercn

**Comment by caseylobrien** [comment]
@javiercn So far I have been able to figure out all the nuances of the Mvc solution so far, but run the Tests (by executing `build.cmd /p:NoBuild=true`) I get the following error: 

`C:\Users\Casey\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\modules\NuGetPackageVerifier\module.targets(37,5): error : SignRequestManifest file F:\OpenSource\Mvc\artifacts\signrequest.xml does not exist. [C:\Users\Casey\.dotnet\buildtools\korebuild\2.2.0-preview1-20180731.1\KoreBuild.proj]`

But at the same time the results are as follows:
`  Total tests: 11938. Passed: 11930. Failed: 0. Skipped: 8.
  Test Run Successful.
  Test execution time: 1.4913 Minutes`

I did check and `signrequest.xml` does not exist. I am not sure what I am missing, any help would be much appreciated.

**Comment by caseylobrien** [comment]
Curious, this morning after running the test suite a few more times the `signrequest.xml` file is now there and the tests run without error. Still skipping 8 tests though. @javiercn @ajaybhargavb Thanks for looking into the error!

**Comment by javiercn** [comment]
This was done as part of 990bed7
